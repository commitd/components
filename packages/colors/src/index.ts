export { presetColors } from './preset'
export type { Color } from './preset'
