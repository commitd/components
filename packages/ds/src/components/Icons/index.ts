export {
  Check,
  CheckIndeterminate,
  ChevronDown,
  ChevronLeft,
  ChevronRight,
  ChevronUp,
  Close,
  DarkMode,
  LightMode,
} from './Icons'
