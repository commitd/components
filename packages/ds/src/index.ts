// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
