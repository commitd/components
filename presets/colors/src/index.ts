export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
