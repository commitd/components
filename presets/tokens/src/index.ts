export { preset as presetTokens } from './preset'
