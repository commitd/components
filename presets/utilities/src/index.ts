export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

