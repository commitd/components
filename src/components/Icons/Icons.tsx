import {
  mdiBrightness3,
  mdiBrightness5,
  mdiChevronDown,
  mdiChevronLeft,
  mdiChevronRight,
  mdiChevronUp,
  mdiClose,
} from '@mdi/js'
import React, { ComponentProps, FC } from 'react'
import type { CSS } from '../../stitches.config'
import { Svg } from '../Svg'

type IconType = FC<ComponentProps<typeof Svg>>

export const Check: IconType = (props) => (
  <Svg
    path="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
    {...props}
  />
)

export const ChevronUp: IconType = (props) => (
  <Svg path={mdiChevronUp} {...props} />
)

export const ChevronDown: IconType = (props) => (
  <Svg path={mdiChevronDown} {...props} />
)

export const ChevronLeft: IconType = (props) => (
  <Svg path={mdiChevronLeft} {...props} />
)

export const ChevronRight: IconType = (props) => (
  <Svg path={mdiChevronRight} {...props} />
)

export const Close: IconType = (props) => <Svg path={mdiClose} {...props} />

export const CheckIndeterminate: IconType = ({ css, ...props }) => (
  <Svg path="M6,11 h12 v4 h-12 Z" css={{ ...css } as CSS} {...props} />
)

export const LightMode: IconType = (props) => (
  <Svg path={mdiBrightness5} {...props} />
)

export const DarkMode: IconType = (props) => (
  <Svg path={mdiBrightness3} {...props} />
)
