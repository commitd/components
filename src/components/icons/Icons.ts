import SvgIcon, { SvgIconProps } from '@material-ui/core/SvgIcon'
import * as MaterialIcons from '@material-ui/icons'

export interface IconProps extends SvgIconProps {}

type SvgIconComponent = typeof SvgIcon

const icons = Object.keys(MaterialIcons).filter(key => key.endsWith('Sharp'))

export const Icons: { [key: string]: SvgIconComponent } = icons.reduce(
  (obj, key) => {
    obj[key.slice(0, -5)] = MaterialIcons[key]
    return obj
  },
  {}
)
