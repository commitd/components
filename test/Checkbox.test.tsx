import React from 'react'
import { renderLight, renderDark, userEvent } from './setupTests'
import { Checkbox, colors, FormControlLabel } from '../src'
import { Icons } from '../stories/util/Icons'

it('renders light without crashing', () => {
  const { asFragment } = renderLight(<Checkbox />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const { asFragment } = renderDark(<Checkbox />)
  expect(asFragment()).toMatchSnapshot()
})

it('expect click to call handler', () => {
  const onClick = jest.fn()
  const { getByRole } = renderLight(<Checkbox onChange={onClick} />)
  userEvent.click(getByRole('checkbox'))
  expect(onClick).toHaveBeenCalled()
})

it('expect click to call handler from click on label', () => {
  const text = 'Test'
  const onClick = jest.fn()
  const { getByText } = renderLight(
    <FormControlLabel
      value="top"
      control={<Checkbox color="primary" />}
      label={text}
      onChange={onClick}
    />
  )
  userEvent.click(getByText(text))
  expect(onClick).toHaveBeenCalled()
})

it('renders variants without crashing', () => {
  const { asFragment } = renderLight(
    <>
      <Checkbox color="default" />
      <Checkbox
        checked={true}
        color="default"
        value="checkedA"
        inputProps={{
          'aria-label': 'checkbox',
        }}
      />
      <Checkbox
        checked={false}
        color="primary"
        inputProps={{
          'aria-label': 'primary checkbox',
        }}
      />
      <Checkbox
        inputProps={{
          'aria-label': 'uncontrolled-checkbox',
        }}
      />
      <Checkbox
        disabled={true}
        inputProps={{
          'aria-label': 'disabled checkbox',
        }}
      />
      <Checkbox
        disabled={true}
        checked={true}
        inputProps={{
          'aria-label': 'disabled checked checkbox',
        }}
      />
      <Checkbox
        indeterminate={true}
        inputProps={{
          'aria-label': 'indeterminate checkbox',
        }}
      />
      <Checkbox
        defaultChecked={true}
        color="default"
        inputProps={{
          'aria-label': 'checkbox with default color',
        }}
      />
      <Checkbox
        style={{ color: 'red' }}
        icon={<Icons.FavoriteBorder />}
        checkedIcon={<Icons.Favorite />}
      />
      <Checkbox
        icon={<Icons.BookmarkBorder style={{ color: colors.grey[600] }} />}
        checkedIcon={
          <Icons.Bookmark style={{ color: colors.lightBlue[400] }} />
        }
      />
    </>
  )
  expect(asFragment()).toMatchSnapshot()
})
