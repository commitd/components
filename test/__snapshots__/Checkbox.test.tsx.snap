// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</DocumentFragment>
`;

exports[`renders variants without crashing 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root PrivateSwitchBase-checked MuiCheckbox-checked"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="checkbox"
        checked=""
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value="checkedA"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          class="makeStyles-tick makeStyles-tick"
          d="M4 6H19V18H4z"
        />
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="primary checkbox"
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="uncontrolled-checkbox"
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="true"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-disabled MuiCheckbox-disabled MuiIconButton-colorSecondary MuiIconButton-disabled MuiButtonBase-disabled"
    tabindex="-1"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="disabled checkbox"
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        disabled=""
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    aria-disabled="true"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked MuiCheckbox-checked PrivateSwitchBase-disabled MuiCheckbox-disabled MuiIconButton-colorSecondary MuiIconButton-disabled MuiButtonBase-disabled"
    tabindex="-1"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="disabled checked checkbox"
        checked=""
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        disabled=""
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorDisabled"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          class="makeStyles-tick makeStyles-tick"
          d="M4 6H19V18H4z"
        />
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-indeterminate MuiIconButton-colorSecondary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="indeterminate checkbox"
        class="PrivateSwitchBase-input"
        data-indeterminate="true"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root PrivateSwitchBase-checked MuiCheckbox-checked"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        aria-label="checkbox with default color"
        checked=""
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          class="makeStyles-tick makeStyles-tick"
          d="M4 6H19V18H4z"
        />
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    style="color: red;"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="color: rgb(98, 98, 98);"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5v18l7-3 7 3V3zm-2 15l-5-2.18L7 18V5h10v13z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</DocumentFragment>
`;
